src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -607,13 +607,21 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
           RequestNewWorkerIfNeeded(scheduling_key);
 
         } else {
-          if (IsRayletFailed(RayConfig::instance().RAYLET_PID())) {
-            RAY_LOG(WARNING)
-                << "The worker failed to receive a response from the local "
-                   "raylet because the raylet is crashed. Terminating the worker.";
-            QuickExit();
+          if (status.IsGrpcUnavailable()) {
```

Comment:
Hmm kind of don't like that we have 2 ways to detect node failure; `IsRayletFailed ` (for local socket) and `status.IsGrpcUnavailable()` for GRPC. Is there any way to consolidate them? Maybe we can change the signature of `IsRayletFailed` to `IsRayletFailed(pid, status)`?

Reply:
Yea, that place also needs to be changed since IsRayletFailed doesn't work for driver but that will be done separately.